app/assets/javascripts/test/backend-snapshot-tests/scripts.e2e.js

```diff
@@ -0,0 +1,53 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
+/* eslint-disable import/first */
+// @flow
+import test from "ava";
+import * as api from "admin/admin_rest_api";
+import { tokenUserA, setCurrToken } from "../enzyme/e2e-setup";
+
+test.before("Change token", async () => {
+  setCurrToken(tokenUserA);
+});
+
+test("getScripts()", async t => {
+  const scripts = await api.getScripts();
+  t.snapshot(scripts, { id: "scripts-getScripts" });
+});
+
+test("getScript()", async t => {
+  const scripts = await api.getScripts();
+  const firstScript = scripts[0];
+
+  const script = await api.getScript(firstScript.id);
+  t.snapshot(script, { id: "scripts-getScript" });
+});
+
+test("createScript(), updateScript(), and deleteScript()", async t => {
+  const activeUser = await api.getActiveUser();
+
+  const data = {
+    id: "will-be-ignored-anyway",
+    name: "MergerMode",
+    owner: activeUser.id,
+    gist: "https://gist.github.com/heikowissler/d5ff5d490ab381af9405c9078096c723",
+  };
+
+  // Create New Script
+  const createdScript = await api.createScript(data);
+
+  // Since the id will change after re-runs, we fix it here for easy
+  // snapshotting
+  const createdScriptWithFixedId = Object.assign({}, createdScript, { id: "fixed-script-id" });
+  t.snapshot(createdScriptWithFixedId, { id: "scripts-createScript" });
+
+  // Update Script
+  const newData = Object.assign({}, createdScript, { name: "MegaScript", owner: activeUser.id });
```

Comment:
Isn't assigning `owner` unnecessary since it's equal to `data.owner`?

Reply:
Nope. The API downloads a complete owner object and the `updateScript` function only requires the `id` of the owner.